### PR TITLE
fix/issue 118- fixed infinite loader in condition table

### PR DESCRIPTION
**Pull request by YashilDepani**
In this PR, I have fixed the infinite loader issue. This issue was occurring in ```Enrollments by Condition```  section whenever a user visits for the first time.

### CODE REVIEW

File: frontend/projects/abtesting/src/app/features/dashboard/home/components/enrollment-condition-table/enrollment-condition-table.component.ts [hunk 1]
```diff
@@ -31,7 +33,7 @@ export class EnrollmentConditionTableComponent implements OnChanges, OnInit {
   }
 
   ngOnInit() {
-    this.experimentService.experimentStatById$(this.experiment.id).subscribe(stat => {
+    this.experimentConditionSub = this.experimentService.experimentStatById$(this.experiment.id).subscribe(stat => {
```
Comment (JD2455) [inline frontend/projects/abtesting/src/app/features/dashboard/home/components/enrollment-condition-table/enrollment-condition-table.component.ts:36]: @YashilDepani Please update the variable name `this.experimentConditionSub` to `this.experimentStateSub`.

**Review by JD2455 — CHANGES_REQUESTED** [review]

File: frontend/projects/abtesting/src/app/features/dashboard/home/components/enrollment-condition-table/enrollment-condition-table.component.ts (hunk 1 above)
Comment (YashilDepani) [reply frontend/projects/abtesting/src/app/features/dashboard/home/components/enrollment-condition-table/enrollment-condition-table.component.ts]: ok, code updated.

**Review by JD2455 — APPROVED** [review]

File: frontend/projects/abtesting/src/app/core/experiments/store/experiments.reducer.ts [hunk 2]
```diff
@@ -166,8 +166,8 @@ const reducer = createReducer(
   on(
     experimentsAction.actionFetchExperimentDetailStatSuccess,
     (state, { stat }) => {
-      state.stats[stat.id] = stat;
-      return state;
+      const updatedStat = state.stats[stat.id] = stat;
```
Comment (msandbothe) [inline frontend/projects/abtesting/src/app/core/experiments/store/experiments.reducer.ts:169]: Although this is valid, this is hard to parse as a one-liner.

Comment (YashilDepani) [reply frontend/projects/abtesting/src/app/core/experiments/store/experiments.reducer.ts]: I have updated the code.
